lib/matplotlib/cbook/__init__.py

```diff
@@ -1367,11 +1367,22 @@ def _check_1d(x):
         return np.atleast_1d(x)
     else:
         try:
-            ndim = x[:, None].ndim
             # work around https://github.com/pandas-dev/pandas/issues/27775
             # which mean the shape is not as expected. That this ever worked
             # was an unintentional quirk of pandas the above line will raise
             # an exception in the future.
+            # This warns in pandas >= 1.0 via
+            # https://github.com/pandas-dev/pandas/pull/30588
+            with warnings.catch_warnings() as w:
+                warnings.filterwarnings("ignore",
+                                        category=DeprecationWarning,
+                                        module='pandas[.*]')
```

Comment:
does this mean we're just kicking the can down the road and will need yet another fix when pandas actually change this behavior?

Reply:
The change will be to raise one of the Exceptions we already catch below so we will still be happy.

Reply:
fair enough